extends CharacterBody2D

@onready var bus_stop = get_node("/root/MainScene/BusStop")  # Absolute path
@export var target_position: Vector2  # ‡∏ï‡∏≥‡πÅ‡∏´‡∏ô‡πà‡∏á‡∏ó‡∏µ‡πà DekKu ‡∏à‡∏∞‡πÄ‡∏î‡∏¥‡∏ô‡πÑ‡∏õ (‡πÄ‡∏ä‡πà‡∏ô ‡∏´‡∏ô‡πâ‡∏≤‡∏£‡∏ñ)

var walking := false
var speed := 100
#var velocity := Vector2.ZERO

func _ready():
	bus_stop = get_tree().get_nodes_in_group("bus_stops").front()
	if bus_stop:
		var is_here = bus_stop.connect("bus_arrived", self._on_bus_arrived)
		print(is_here)
		print("üë¶ DekKu ‡∏û‡∏£‡πâ‡∏≠‡∏°‡∏£‡∏±‡∏ö signal ‡πÅ‡∏•‡πâ‡∏ß")
	else:
		print("‚ùå DekKu ‡∏¢‡∏±‡∏á‡πÑ‡∏°‡πà‡πÑ‡∏î‡πâ‡πÄ‡∏ä‡∏∑‡πà‡∏≠‡∏° bus_stop ‡πÉ‡∏ô Inspector")

func _on_bus_arrived():
	print("üë¶ DekKu: ‡πÑ‡∏î‡πâ‡∏£‡∏±‡∏ö signal ‡πÅ‡∏•‡πâ‡∏ß! ‡πÄ‡∏£‡∏¥‡πà‡∏°‡πÄ‡∏î‡∏¥‡∏ô‡πÑ‡∏õ‡∏ó‡∏µ‡πà‡∏£‡∏ñ")
	walking = true

func _physics_process(delta):
	if bus_stop:
		var is_here = bus_stop.connect("bus_arrived", self._on_bus_arrived)
		print(is_here)
		print("üë¶ DekKu ‡∏û‡∏£‡πâ‡∏≠‡∏°‡∏£‡∏±‡∏ö signal ‡πÅ‡∏•‡πâ‡∏ß")
	else:
		print("‚ùå DekKu ‡∏¢‡∏±‡∏á‡πÑ‡∏°‡πà‡πÑ‡∏î‡πâ‡πÄ‡∏ä‡∏∑‡πà‡∏≠‡∏° bus_stop ‡πÉ‡∏ô Inspector")
		

	if walking:
		var direction = (target_position - global_position).normalized()
		velocity = direction * speed
		move_and_slide()

		# Check if reached target
		if global_position.distance_to(target_position) < 5:
			walking = false
			velocity = Vector2.ZERO
			print("‚úÖ DekKu: ‡∏Ç‡∏∂‡πâ‡∏ô‡∏£‡∏ñ‡πÄ‡∏£‡∏µ‡∏¢‡∏ö‡∏£‡πâ‡∏≠‡∏¢‡πÅ‡∏•‡πâ‡∏ß")
